substrate/frame/revive/src/lib.rs

```diff
@@ -267,6 +267,12 @@ pub mod pallet {
 			BlockNumberFor<Self>,
 		>;
 
+		/// The weigher used to calculate XCM execution costs
+		#[pallet::no_default_bounds]
+		type XcmWeigher: xcm_executor::traits::WeightBounds<
+			<Self as frame_system::Config>::RuntimeCall,
+		>;
```

Comment:
Can't we just add the bound to the `Xcm` associated type? It is all implemented by `pallet_xcm` anyways AFAIK.

Reply:
No, this is currently a runtime configuration object, also being set to the xcm-executor, not pallet-xcm. We should consolidate it somehow to not have to set it twice (pallet-revive and xcm-executor), but this is not trivial and “nice-to-have” at this point IMO